Rename (again) *widthFor* functions in FontCascade keeping Simple/Complex names

vitorroriz opened the pull request:
#### 5cb9d01d4dd6832731502e5e14951d25da3dfe7d
<pre>
Rename (again) *widthFor* functions in FontCascade keeping Simple/Complex names
<a href="https://bugs.webkit.org/show_bug.cgi?id=272831">https://bugs.webkit.org/show_bug.cgi?id=272831</a>
&lt;<a href="https://rdar.apple.com/problem/126631132">rdar://problem/126631132</a>&gt;

Reviewed by Fujii Hironori.

After our recent rename to these functions at 31aadb040e324bd558d1b40cf825e1f7d6d153c2,
we got feedback requesting that we keep the &quot;Simple&quot;/&quot;Complex&quot;
names in the meaning, since it is consistently used along FontCascade.

* Source/WebCore/platform/graphics/FontCascade.cpp:
(WebCore::FontCascade::width const):
(WebCore::FontCascade::widthForSimpleText const):
(WebCore::FontCascade::widthForComplexText const):
(WebCore::FontCascade::offsetForPositionForSimpleText const):
(WebCore::FontCascade::widthForTextUsingWidthIterator const): Deleted.
(WebCore::FontCascade::widthForTextUsingComplexTextController const): Deleted.
* Source/WebCore/platform/graphics/FontCascade.h:

Canonical link: <a href="https://commits.webkit.org/277634@main">https://commits.webkit.org/277634@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/aee899667d42abc0540b53f2937726e22767e0ba

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/48146 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/27355 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/51085 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/50835 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/44212 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/50452 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/33290 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/24862 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/39348 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/48728 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/25067 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/41633 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/20476 "Found 1 new API test failure: /WPE/TestLoaderClient:/webkit/WebKitWebView/load-twice-and-reload (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/22545 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/42809 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/6202 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/44520 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/43252 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/52736 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/23193 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/19564 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/46682 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/24459 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/41802 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/45575 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10624 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/25263 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/24181 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

fujii reviewed (approved):
Very nice. Thank you very much.

vitorroriz committed:
Rename (again) *widthFor* functions in FontCascade keeping Simple/Complex names
https://bugs.webkit.org/show_bug.cgi?id=272831
<rdar://problem/126631132>

Reviewed by Fujii Hironori.

After our recent rename to these functions at 31aadb040e324bd558d1b40cf825e1f7d6d153c2,
we got feedback requesting that we keep the "Simple"/"Complex"
names in the meaning, since it is consistently used along FontCascade.

* Source/WebCore/platform/graphics/FontCascade.cpp:
(WebCore::FontCascade::width const):
(WebCore::FontCascade::widthForSimpleText const):
(WebCore::FontCascade::widthForComplexText const):
(WebCore::FontCascade::offsetForPositionForSimpleText const):
(WebCore::FontCascade::widthForTextUsingWidthIterator const): Deleted.
(WebCore::FontCascade::widthForTextUsingComplexTextController const): Deleted.
* Source/WebCore/platform/graphics/FontCascade.h:

Canonical link: https://commits.webkit.org/277634@main
